pkg/types/hashedrekord/hashedrekord.go

```diff
@@ -0,0 +1,40 @@
+// Copyright 2025 The Sigstore Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package hashedrekord
+
+import (
+	"fmt"
+
+	pb "github.com/sigstore/rekor-tiles/pkg/generated/protobuf"
+)
+
+func Validate(hr *pb.HashedRekordRequest) error {
+	if len(hr.Signature) == 0 {
+		return fmt.Errorf("missing signature")
+	}
+	if hr.Verifier == nil {
+		return fmt.Errorf("missing verifier")
+	}
+	if hr.Verifier.GetPublicKey() == nil && hr.Verifier.GetX509Certificate() == nil {
```

Comment:
Old hashedrekord didn't have separate fields for public key and x.509 cert https://github.com/sigstore/rekor/blob/4b01f831f8b23f9070f59f4e476e3eabc5b68ab5/pkg/generated/models/hashedrekord_v001_schema.go#L394